tests/core/test_utilities.py

```diff
@@ -320,6 +320,20 @@ def test_voxelize_volume_no_face_mesh(rectilinear):
         pv.voxelize_volume(pv.PolyData())
 
 
+@pytest.mark.parametrize('function', [pv.voxelize_volume])
+def test_voxelize_enclosed_bounds(function):
+    mesh = ex.load_ant()
+    vox = function(mesh, density=1, enclosed=True)
+
+    assert (vox.bounds.x_min <= mesh.bounds.x_min)
+    assert (vox.bounds.y_min <= mesh.bounds.y_min)
+    assert (vox.bounds.z_min <= mesh.bounds.z_min)
+
+    assert (vox.bounds.x_max >= mesh.bounds.x_max)
+    assert (vox.bounds.y_max >= mesh.bounds.y_max)
+    assert (vox.bounds.z_max >= mesh.bounds.z_max)
```

Comment:
This change was made because there was missing test coverage for `pv.voxelize`. But now the tests are failing:
https://github.com/pyvista/pyvista/actions/runs/10548641482/job/29222787144#step:12:6076

This must be related to this https://github.com/pyvista/pyvista/pull/6514#discussion_r1726604581:

> Also with this test - I realised it isn't a good test for voxelize as depending on the shape of the mesh, the outer cells can be discarded. This is true for the ant example. The test is good for voxelize_volume though and since they both use the same underlying binning function the base grid generated is the same.

@aeverallpx can you elaborate on this? What do you mean by outer cells can be discarded,  and why is this OK/expected?

We can use a different fixture just to make the tests pass but I think the term `enclosed` should imply that this test should always pass. Is there something I'm missing about how this option works?

Reply:
Sorry it has taken a while to get back to this. 

The problem was that voxelize choses vertices to include if they are contained in the input mesh surface. 
```selection = ugrid.select_enclosed_points(surface, tolerance=0.0, check_surface=check_surface)```

I've  changed this to include cells which are within one cell of the surface which means that the input mesh is enclosed by the produced voxelization. Below are screenshots for voxelization of the ant with `enclosed=False` and `enclosed=True`.
<img width="952" alt="image" src="https://github.com/user-attachments/assets/89f2dc10-38fc-49ab-a123-78093e17eff1">

Tests now all pass. I hope this resolves the issue.